### PR TITLE
Stojanovic/ep 92 community landing page

### PR DESCRIPTION
**Pull request by Ducica**
Adding UI to oarepo-communities
Transferring needed routes to be under our blueprint instead of invenio-communities

-----------
The UI for invitations will come in a separate PR
There is an issue with the URL that takes you to the form, as most of the templates are rendered on their end, we don't have access to extra_context and similar variables, therefore, not sure what to do with this.
image

Related PRs 

https://github.com/oarepo/invenio-communities/pull/1 (this one is critical for the functioning)

https://github.com/oarepo/oarepo-communities/pull/66/files#diff-01617dd0f2d5c6fa294b95b8c0ad633d42d2ca9b5b46868ffbacdf785dbe22ff

https://github.com/Narodni-repozitar/nr-docs/pull/157

For sure, there shall more things to iron out, but the PR is getting a bit difficult to manage due to size, so please review and hopefully after any corrections we could merge.

### CODE REVIEW

File: oarepo_communities/ui/communities_components/templates/oarepo/invenio_communities/details/header.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{% extends "semantic-ui/invenio_communities/details/header.html" %}
+
+{%- block subheader_buttons %}
+    <div
+    class="sixteen wide mobile sixteen wide tablet three wide computer right aligned middle aligned column">
+    {%- if not community_use_jinja_header %}
+        {# TODO: Need to figure out what to do about the url as we dont have access 
+        to our input as templates are rendered on their end.
+         #}
+        <a href="/docs/_new?community={{ community.slug }}"
```
Comment (mesemus) [inline oarepo_communities/ui/communities_components/templates/oarepo/invenio_communities/details/header.html:10]: For the time being, use settings and format?
```
<a href="{{ config.NEW_RECORD_PAGE.format(community_slug=community.slug) }}"
```

or

```
<a href="{{ config.RECORD_ROUTES['new'].format(community_slug=community.slug) }}"
```
(in sync with config.COMMUNITY_ROUTES used on frontpage.html)

File: oarepo_communities/ui/oarepo_communities/__init__.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+    ):
+        return invenio_communities_settings_pages(
+            pid_value=resource_requestctx.view_args["pid_value"]
+        )
+
+    @request_view_args
+    def members(
+        self,
+    ):
+        print(g.identity, "identityflask", flush=True)
```
Comment (mesemus) [inline oarepo_communities/ui/oarepo_communities/__init__.py:131]: remove this

File: oarepo_communities/ui/oarepo_communities/templates/CommunityRecordPage.jinja [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{#def  extra_context, search_app_config #}
```
Comment (mesemus) [inline oarepo_communities/ui/oarepo_communities/templates/CommunityRecordPage.jinja:1]: rename the file to CommunityRecordsPage (plural?)

**Review by mesemus — CHANGES_REQUESTED** [review]

File: oarepo_communities/i18n/translations/cs/LC_MESSAGES/messages.po [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+#: /home/dusanst/Projects/oarepo-dashboard/oarepo_dashboard/ui/dashboard_records/__init__.py:58
+msgid "User Dashboard"
+msgstr "Nástěnka"
+
+#: /home/dusanst/Projects/oarepo-communities/oarepo_communities/ui/communities_landing/templates/CommunityRecordPage.jinja:2
+msgid "Community"
+msgstr ""
+
+#: /home/dusanst/Projects/oarepo-communities/oarepo_communities/ui/communities_landing/templates/CommunityRecordPage.jinja:2
+msgid "NR Document repository"
```
Comment (mirekys) [inline oarepo_communities/i18n/translations/cs/LC_MESSAGES/messages.po:142]: Shouldnt be here

File: oarepo_communities/ui/communities_components/semantic-ui/less/oarepo_communities/themes/default/views/card.variables [hunk 5]
```diff
@@ -0,0 +1,2 @@
+@largeDesktopScreenCardHeight: 21.76rem;
```
Comment (mirekys) [inline oarepo_communities/ui/communities_components/semantic-ui/less/oarepo_communities/themes/default/views/card.variables:1]: this should probably be pixel units, images usually dont scale relative to font size

File: oarepo_communities/ui/oarepo_communities/templates/CommunityRecordsPage.jinja [hunk 6]
```diff
@@ -0,0 +1,21 @@
+{#def  extra_context, search_app_config #}
+{%- set title = _("Community") ~ " " ~ request.path.split('/')[-1] ~ " | " ~ _("NR Document repository") %}
```
Comment (mirekys) [inline oarepo_communities/ui/oarepo_communities/templates/CommunityRecordsPage.jinja:2]: There should be no reference to nr docs

Comment (mirekys) [inline oarepo_communities/ui/oarepo_communities/templates/CommunityRecordsPage.jinja:2]: also community name shouldnt be parsed from url like this, cant we use somehing better coming from context props?

File: oarepo_communities/ui/communities_components/templates/oarepo/invenio_communities/search.html [hunk 7]
```diff
@@ -0,0 +1,4 @@
+    {% extends "semantic-ui/invenio_communities/search.html" %}
```
Comment (mirekys) [inline oarepo_communities/ui/communities_components/templates/oarepo/invenio_communities/search.html:1]: Is this template necessary?

**Review by mirekys — CHANGES_REQUESTED** [review]

File: oarepo_communities/ui/communities_components/semantic-ui/less/oarepo_communities/themes/default/views/card.variables (hunk 5 above)
Comment (Ducica) [reply oarepo_communities/ui/communities_components/semantic-ui/less/oarepo_communities/themes/default/views/card.variables:1]: These are some communities styles that were in RDM, so I thought I would place them in oarepo-communities in our case. It was a copy paste. Converting to pixels.

File: oarepo_communities/ui/communities_components/templates/oarepo/invenio_communities/search.html (hunk 7 above)
Comment (Ducica) [reply oarepo_communities/ui/communities_components/templates/oarepo/invenio_communities/search.html:1]: @mirekys Sorry my bad. Previously I had overrided a new community button there, but now the template is indeed useless.

**Review by mirekys — APPROVED** [review]

**Review by mesemus — APPROVED** [review]

**Review by mirekys — APPROVED** [review]

**Pull request merged**